source/language/gates.rst

```diff
@@ -289,9 +289,11 @@ The modifier ``ctrl @`` replaces its gate argument :math:`U` by a
 controlled-:math:`U` gate. If the control bit is 0, nothing happens to the target bit.
 If the control bit is 1, :math:`U` acts on the target bit. Mathematically, the controlled-:math:`U`
 gate is defined as :math:`C_U = I \otimes U^c`, where :math:`c` is the integer value of the control
-bit and :math:`C_U` is the controlled-:math:`U` gate. The new control qubit is prepended to the
-argument list for the controlled-:math:`U` gate. The modified gate does not use any additional
-scratch space and may require compilation to be executed.
+bit and :math:`C_U` is the controlled-:math:`U` gate. The control modifier argument can be a
+register, and in this case controlled gate broadcast over it (as it shown in examples above for
+CNOT gate). The new control argument is prepended to the argument list for the
```

Comment:
I think the previous terminology of "new control qubit" was slightly clearer, because it doesn't clash with the value of `n` - you mentioned it in your main post, but it's quite a subtle distinction that I think will just be a bit harder to understand for the reader.  We could say "quantum argument" if you prefer not to say "qubit" (since it can also be a collection of qubits).

I'd put the bit about "the new control quantum argument is prepended to the argument list" back in its original place, and put your new sentence about "The new arguments may also be quantum registers, in which case the standard broadcasting rules apply" after - we want to put the most important, common circumstances first, and handle the special cases later.

Reply:
Done